File: legacy/src/main/java/com/redbox/domain/user/repository/EmailVerificationCodeRepository.java
package com.redbox.domain.user.repository;

import lombok.RequiredArgsConstructor;
import org.springframework.data.redis.core.StringRedisTemplate;
import org.springframework.data.redis.core.ValueOperations;
import org.springframework.stereotype.Repository;

import java.time.Duration;
import java.util.Optional;

@Repository
@RequiredArgsConstructor
public class EmailVerificationCodeRepository {

    private final StringRedisTemplate stringRedisTemplate;

    public void save(String key, String value) {
        ValueOperations<String, String> valueOperations = stringRedisTemplate.opsForValue();
        Duration expireDuration = Duration.ofMinutes(5);

        valueOperations.set(key, value, expireDuration);
    }

    public Optional<String> getVerificationCodeByEmail(String key) {
        return Optional.ofNullable(stringRedisTemplate.opsForValue().get(key));
    }

    public void deleteByEmail(String key) {
        stringRedisTemplate.delete(key);
    }
}

File: migration/src/main/kotlin/com/redbox/domain/user/user/repository/EmailVerificationCodeRepository.kt
package com.redbox.domain.user.user.repository

import org.springframework.data.redis.core.StringRedisTemplate
import org.springframework.data.redis.core.ValueOperations
import org.springframework.stereotype.Repository
import java.time.Duration

@Repository
class EmailVerificationCodeRepository(
    private val stringRedisTemplate: StringRedisTemplate
) {
    fun save(key: String, value: String) {
        val valueOperations: ValueOperations<String, String> = stringRedisTemplate.opsForValue()
        val expireDuration: Duration = Duration.ofMinutes(5)

        valueOperations.set(key, value, expireDuration)
    }

    fun getVerificationCodeByEmail(key: String): String? {
        return stringRedisTemplate.opsForValue()[key]
    }

    fun deleteByEmail(key: String) {
        stringRedisTemplate.delete(key)
    }
}